ts/src/header-validator/validate-json.ts

```diff
@@ -1279,28 +1279,54 @@ function aggregatableTriggerData(
   )
 }
 
+export type AggregatableValuesValue =
+  | {
+      value: number
+      filteringId: bigint
+    }
+  | number
+
 export type AggregatableValuesConfiguration = FilterPair & {
-  values: Map<string, number>
+  values: Map<string, AggregatableValuesValue>
+}
+
+function aggregatableKeyValueValue(ctx: Context, j: Json): Maybe<number> {
+  return number(ctx, j)
+    .filter((n) => isInteger(ctx, n))
+    .filter((n) =>
+      isInRange(ctx, n, 1, constants.allowedAggregatableBudgetPerSource)
+    )
 }
 
 function aggregatableKeyValue(
   ctx: Context,
   [key, j]: [string, Json]
-): Maybe<number> {
+): Maybe<AggregatableValuesValue> {
   if (!aggregationKeyIdentifierLength(ctx, key, 'key ')) {
     return None
   }
-  return number(ctx, j)
-    .filter((n) => isInteger(ctx, n))
-    .filter((n) =>
-      isInRange(ctx, n, 1, constants.allowedAggregatableBudgetPerSource)
-    )
+
+  return typeSwitch(ctx, j, {
+    number: (ctx, j) => aggregatableKeyValueValue(ctx, j),
+    object: (ctx, j) =>
+      struct(ctx, j, {
+        value: field('value', aggregatableKeyValueValue),
+        filteringId: field('filtering_id', uint64, null),
```

Comment:
Here we are using `uint64`, but the spec currently just says:

> Set filteringId to the result of applying the [rules for parsing non-negative integers](https://html.spec.whatwg.org/multipage/common-microsyntaxes.html#rules-for-parsing-non-negative-integers) to value["filtering_id"].

Maybe the spec should use a sized integer type?

Reply:
64 bit is the maximum size for a filtering id (assuming that max bytes is set to 8). The actual limit is validated later in `areFilteringIdsWithinMaxBytes` when the max bytes property has been parsed.

In the spec, the next line validates the actual limit. I think it's clearer to validate the size directly on the actual limit.

For the validator, changing the structure to pass the limit did not seem like it'd worth the effort. In practice, the worst case is that they'd hit this limit but the actual limit is even lower (due to having a lower configured max bytes). The optimal solution being that we check on the actual limit immediately

Reply:
Ack.

Reply:
After reviewing, `areFilteringIdsWithinMaxBytes` also had some complexities and was fragile given that it had to "replicate" the scope. As such, I updated to pass down the maxBytes value making the validator implementation closer to the spec.